### PR TITLE
Add optional icons to profile menu link items in the navbar

### PR DESCRIPTION
**Pull request by acomsto**
## Description

- Added the option to add an icon to the link items within the profile menu on the navbar.
- Updated Storybook to include added optional icon property on the `ModusNavbarProfileMenuLink` where all the interfaces are listed.

References #1770

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] Documentation update

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules
- [x] I have checked my code and corrected any misspellings

### CODE REVIEW

File: package-lock.json
Comment (coliff) [inline package-lock.json:1]: this file isn't needed.

**Review by msankaran0712 — CHANGES_REQUESTED** [review]
1. Please update the type information in Storybook, 
image

2. Please add a few example links with icon here too, It will be easier for to visualize the changes,
image

File: stencil-workspace/src/components/modus-navbar/profile-menu/modus-navbar-profile-menu.tsx [hunk 1]
```diff
@@ -40,6 +41,11 @@ export class ModusNavbarProfileMenu {
             {this.links.map((link) => {
               return (
                 <div class="link" onClick={() => this.linkClick.emit(link.id)}>
+                  {link.icon && (
```
Comment (msankaran0712) [inline stencil-workspace/src/components/modus-navbar/profile-menu/modus-navbar-profile-menu.tsx:44]: Is it possible to add any unit test for the profile menu popup?

**Comment by cjwinsor** [comment]
The link element is extending outside the profile menu
image

File: stencil-workspace/src/components/modus-navbar/profile-menu/modus-navbar-profile-menu.tsx [hunk 2]
```diff
@@ -40,6 +41,11 @@ export class ModusNavbarProfileMenu {
             {this.links.map((link) => {
```
Comment (msankaran0712) [inline stencil-workspace/src/components/modus-navbar/profile-menu/modus-navbar-profile-menu.tsx:41]: @acomsto I discussed the menu with @coliff. We believe it should be a borderless `modus-list` with links, such as https://modus-bootstrap.netlify.app/docs/v2/components/list-group/#links-and-buttons, and follow the same padding & spacing as the List guidelines https://modus.trimble.com/components/web/lists/styles/ and have a hand cursor
image.

If you can add the borderless support for `modus-list` component, you can use or create a div list following the Modus guidelines for Lists.

@coliff Please correct me if am wrong.

cc @ryan-henness-trimble

**Comment by msankaran0712** [comment]
@acomsto did you get the prototype from @egunther39 ?

**Comment by egunther39** [comment]
@msankaran0712 - I didn't get to it yet. I'll take care of it this morning

**Comment by acomsto** [comment]
image


@egunther39 Looks like the list items and navbar menu backgrounds have different requirements for the dark mode theme. Should I just match the backgrounds to the list item background colors in dark mode?

**Comment by msankaran0712** [comment]
> image
> 
> @egunther39 Looks like the list items and navbar menu backgrounds have different requirements for the dark mode theme. Should I just match the backgrounds to the list item background colors in dark mode?

Stay tuned for another mockup for the dark theme. Meanwhile can you please change background color for all the popups in the Navbar to use `  --modus-gray-10: #171c1e;` for the dark mode, currently it is using a wrong color. Let me know if you need any help in how to do that, because you need to insert a new CSS variable here below and update it in all the popups in Navbar, profile menu pop-up, notifications pop-up, dynamic icon button pop-up, hamburger menu, etc.
image

**Review by msankaran0712 — APPROVED** [review]
@acomsto Thank you for the PR! The UI looks, just a few minor UI corrections needed,

- Dark theme - Pop up border color has to be Gray 8 (#464B52)
- Spacing between the leading icon and the text inside a list item is 16px ([Figma link](https://www.figma.com/file/wyfVJUHWRMkeCfdB38HFEE/Modus---Web?node-id=27038%3A32416&mode=dev))
- Yet to be confirmed, the leading icon size inside a list item is supposed to be 16px not 24px

**Review by msankaran0712 — CHANGES_REQUESTED** [review]
@acomsto Thank you for the PR! The UI looks good, just a few minor UI corrections needed,

Dark theme - Pop up border color has to be Gray 8 (#464B52)
Spacing between the leading icon and the text inside a list item is 16px ([Figma link](https://www.figma.com/file/wyfVJUHWRMkeCfdB38HFEE/Modus---Web?node-id=27038%3A32416&mode=dev))
Yet to be confirmed, the leading icon size inside a list item is supposed to be 16px not 24px

**Comment by msankaran0712** [comment]
@acomsto Kindly comment out the failing test 
image

**Comment by msankaran0712** [comment]
@acomsto, is there anything else pending from Modus' end?

**Comment by acomsto** [comment]
@msankaran0712 I don't believe so

**Comment by msankaran0712** [comment]
> @msankaran0712 I don't believe so

Okay, I will be waiting for the PR comments to get fixed

**Comment by msankaran0712** [comment]
@acomsto Kindly squash the commits.

**Comment by ryan-henness-trimble** [comment]
> @acomsto Kindly squash the commits.

@msankaran0712 
Should we allow devs to depend on the "Squash and merge" GitHub feature instead of requiring explicit squash?

**Comment by msankaran0712** [comment]
> > @acomsto Kindly squash the commits.
> 
> @msankaran0712 Should we allow devs to depend on the "Squash and merge" GitHub feature instead of requiring explicit squash?

I am okay with the GitHub feature. Explicitly squashing commits is optional, and we can do it if it is not too much effort.

**Review by msankaran0712 — CHANGES_REQUESTED** [review]

**Comment by msankaran0712** [comment]
@acomsto @ryan-henness-trimble  Apologies, I did not submit my review comments until now thinking that I already did.

File: stencil-workspace/src/components/modus-list-item/modus-list-item.tsx [hunk 3]
```diff
@@ -1,24 +1,31 @@
...
+
   /** (optional) The selected state of the list item */
   @Prop() selected: boolean;
 
   /** (optional) The size of list item */
   @Prop() size: 'condensed' | 'large' | 'standard' = 'standard';
 
   /** (optional) The type of list item */
-  @Prop() type = 'standard'; // Future support for 'checkbox' | 'icon' | 'menu' | 'standard' | 'switchLeft' | 'switchRight'
+  @Prop() type: 'standard' | 'icon'; // Future support for 'checkbox' | 'icon' | 'menu' | 'standard' | 'switchLeft' | 'switchRight'
```
[the remark this answers is not in the document]
Comment (ryan-henness-trimble) [reply stencil-workspace/src/components/modus-list-item/modus-list-item.tsx:28]: In spirit of keeping this component generic and not assuming what the consumer would like to render in the list item, I don't think we should add the trailing and leading icons. I think the best approach is to add an `icon` property to the `ModusNavbarProfileMenuLink`. That way, we can add the icon within the list item next to its display text using the `<IconMap />`

File: stencil-workspace/src/components/modus-navbar/profile-menu/modus-navbar-profile-menu.spec.ts [hunk 4]
```diff
@@ -0,0 +1,74 @@
+import { newSpecPage } from '@stencil/core/testing';
+import { ModusNavbarProfileMenu } from './modus-navbar-profile-menu';
+
+describe('modus-navbar-profile-menu', () => {
+  beforeAll(() => {
+    class MutationObserverMock {
```
Comment (msankaran0712) [inline stencil-workspace/src/components/modus-navbar/profile-menu/modus-navbar-profile-menu.spec.ts:6]: why do we need to mock a Mutation observer for profile menu popup?

Comment (msankaran0712) [reply stencil-workspace/src/components/modus-navbar/profile-menu/modus-navbar-profile-menu.spec.ts:6]: Also, the padding between the icon and the text is 16px, not 8px, ([Figma](https://www.figma.com/file/wyfVJUHWRMkeCfdB38HFEE/Modus---Web?node-id=27038%3A32394&mode=dev) for reference) It was incorrect on Modus Styleguide and raised a separate issue for fixing that.

Comment (msankaran0712) [reply stencil-workspace/src/components/modus-navbar/profile-menu/modus-navbar-profile-menu.spec.ts:6]: @ryan-henness-trimble Please feel free to approve and merge the PR when the above two comments are fixed.

**Review by ryan-henness-trimble — APPROVED** [review]

**Comment by ryan-henness-trimble** [comment]
@msankaran0712 Thanks for the help on this PR! 😄

**Pull request merged**